R/pipe_consistency_linter.R

```diff
@@ -0,0 +1,83 @@
+#' Pipe consistency linter
+#'
+#' Check that pipe operators are used consistently by file, or optionally
+#' specify one valid pipe operator.
+#'
+#' @param pipe Which pipe operator is valid (either `"%>%"` or `"|>"`). By default
+#' (`"auto"`), the linter has no preference but will check that each file uses
+#' only one type of pipe operator.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "1:3 |> mean() %>% as.character()",
+#'   linters = pipe_consistency_linter()
+#' )
+#'
+#' lint(
+#'   text = "1:3 %>% mean() %>% as.character()",
+#'   linters = pipe_consistency_linter("|>")
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "1:3 %>% mean() %>% as.character()",
+#'   linters = pipe_consistency_linter()
+#' )
+#'
+#' lint(
+#'  text = "1:3 |> mean() |> as.character()",
+#'  linters = pipe_consistency_linter()
+#' )
+#' @evalRd rd_tags("pipe_consistency_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+pipe_consistency_linter <- function(pipe = c("auto", "%>%", "|>")) {
```

Comment:
Question in the wake of #2046 -- what should be done about other magrittr pipes besides `%>%`? It might be strange to swap `%>%` for `|>` if the input uses `%T>%`, `%!>%`, etc, no? Since those don't really have native equivalents, maybe `%>%` should be encouraged for consistency in such cases?

Reply:
I would say they should be counted as a magrittr pipe, so lint them as well if `|>` is mandated, or treat them as a `%>%` ocurrence if set to `"auto"`, WDYT?

Reply:
That makes sense to me. @bairdj you can use this to help:

https://github.com/r-lib/lintr/blob/430123122ca91c4642d21d32ca9a993918a8c1cf/R/shared_constants.R#L219